Fix lacrosse view sensor units

killercode opened the pull request:
There was a mistake on the units from the sensor that were messing up the device readings on rain, windchill and feels like sensors


## Breaking change
No breaking change unless there are specific settings based on region. Software will not break, but readings may be incorrect.


## Proposed change
Readings where incorrect because of wrong units on the source code


## Type of change


- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [ ] New feature (which adds functionality to an existing integration)
- [ ] Deprecation (breaking change to happen in the future)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
- This PR fixes or closes issue: fixes #106148

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [X] I have followed the [development checklist][dev-checklist]
- [X] I have followed the [perfect PR recommendations][perfect-pr]
- [X] The code has been formatted using Ruff (`ruff format homeassistant tests`)
- [X] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] For the updated dependencies - a link to the changelog, or at minimum a diff between library versions is added to the PR description.
- [ ] Untested files have been added to `.coveragerc`.

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone+-status%3Afailure


[dev-checklist]: https://developers.home-assistant.io/docs/development_checklist/
[manifest-docs]: https://developers.home-assistant.io/docs/creating_integration_manifest/
[quality-scale]: https://developers.home-assistant.io/docs/integration_quality_scale_index/
[docs-repository]: https://github.com/home-assistant/home-assistant.io
[perfect-pr]: https://developers.home-assistant.io/docs/review-process/#creating-the-perfect-pr

killercode committed:
Fixed Unit values from sensor.py

There was a mistake on the units from the sensor that where messing up the device readings on rain, windchill and feels like sensors

IceBotYT commented:
Thank you! Unfortunately I replaced my LaCrosse equipment a while back so I can't test this.

killercode commented:
> Thank you! Unfortunately I replaced my LaCrosse equipment a while back so I can't test this.

No problem. My concern is that if you have regional settings and this changes the units reported by the API :/
But I found some reddit users complaining about the Fahrenheit values and rain as well...

Otherwise it would be nice to have a way to override the hardcoded units configuration.

I don't have enough knowledge on how HA works to allow this, and lately I've been very short on time but with some guidance I can improve this solution.

killercode commented:
> Thank you! Unfortunately I replaced my LaCrosse equipment a while back so I can't test this.

No problem. My concern is that if you have regional settings and this changes the units reported by the API :/
But I found some reddit users complaining about the Fahrenheit values and rain aswell...

Otherwise it would be nice to have a way to override the hardcoded units configuration.

I don't have enough knowledge on how HA works to allow this, and lately I've been very short on time but with some guidance I can improve this solution.

IceBotYT commented:
Yes there is a way to modify the recommended units!

I think the returned values do depend on the region, because they work fine for me 😞

Here's how you can change the
units :) 

![image](https://github.com/home-assistant/core/assets/34712694/e318a2f3-6cb9-4cb2-962b-2c7f664150b8)
![image](https://github.com/home-assistant/core/assets/34712694/13e0598f-f299-43c1-a823-d9b55a3a2a7d)

Hope this helps 😁

killercode commented:
> Yes there is a way to modify the recommended units!
> 
> I think the returned values do depend on the region, because they work fine for me 😞
> 
> Here's how you can change the units :)
> 
> ![image](https://private-user-images.githubusercontent.com/34712694/292322752-e318a2f3-6cb9-4cb2-962b-2c7f664150b8.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxOTQ1ODcsIm5iZiI6MTcwMzE5NDI4NywicGF0aCI6Ii8zNDcxMjY5NC8yOTIzMjI3NTItZTMxOGEyZjMtNmNiOS00Y2IyLTk2MmItMmM3ZjY2NDE1MGI4LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQyMTMxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yNjNlNGMxZDc3OWNlZWVjYmYxNmUxYzkxMDRjYmJhZWU4MDBmNWZlOTYwYTBkMjJhM2ZmMjlmNTE1ZjRjODcwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.3xLIaW35jlH3J9ffJ2QCvohfV1459iQD17QgdNX1PPI) ![image](https://private-user-images.githubusercontent.com/34712694/292322762-13e0598f-f299-43c1-a823-d9b55a3a2a7d.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMxOTQ1ODcsIm5iZiI6MTcwMzE5NDI4NywicGF0aCI6Ii8zNDcxMjY5NC8yOTIzMjI3NjItMTNlMDU5OGYtZjI5OS00M2MxLWE4MjMtZDliNTVhM2EyYTdkLmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBSVdOSllBWDRDU1ZFSDUzQSUyRjIwMjMxMjIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDIzMTIyMVQyMTMxMjdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01MTdiNjU2NjE2YzM4MzhkOTQ2YzdlMDE4NDM3MzY2ZTVmNDEyMDcyNmY3YTUzOTQ2Njg2ZDYwNTAxYjBkNDQ4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.uQN55AWF-8xPhugBH5AeSD7e3m3KaReQTramYXTaNDE)
> 
> Hope this helps 😁

:(
That didn't work in the past for me... it was displaying ºC but showing the numerical value in Fahrneit!

I am testing the API right now! I changed my units on my application and I am going to call the API to see if anything changed...

killercode commented:
and btw. Temperature was working nice... only windchill, rain and heat index were wrong

IceBotYT commented:
I'm glad to hear that you're testing the API, I hope it works!

In the meantime, could you let me know what was displaying Fahrenheit instead of Celsius? If it was a dashboard card it could be an option on the card

killercode commented:
The same cards you showed on the above...

Example, I was getting the value 50 for feels like
In the card was displayed as 50 ºC
When I changed to Fahrenheit it was displaying 122 F

Now that I changed it to be ºC in the code, it appears 10 ºC which is 50.9 F which is correct.

In the meanwhile so far tests with API are ok!

I will try to force united states to see if it changes the values from API

killercode commented:
Perfect...  I changed the regional settings on lacross app to put everything opposed to my regional settings, still get good readings on all sensors.

Also I did direct calls to the API and forced the region to be United States, and still the answer has always the same format and data still correct.

Productive Version:
![image](https://github.com/home-assistant/core/assets/2145778/b627cea1-43be-4276-8460-11d49422cfcb)

My Fixed Version
![image](https://github.com/home-assistant/core/assets/2145778/e6546fc1-f8b7-4492-a4b5-30ff3e3e340e)

All seems working nice :D now I am confident!

MartinHjelmare reviewed (approved):
Thanks!

doggkruse commented:
I'm having this issue as well.  I see in this PR it's just changing the hard coded units.  The API actually includes the units and seems like a more robust fix would be to use the value from the API.

![image](https://github.com/home-assistant/core/assets/6468053/5147db93-5f9e-4fac-b3eb-9172669a1d0b)
